### PR TITLE
Move add-cloud-metadata to ECS

### PR DESCRIPTION
**Pull request by ruflin**
Fields renamed:

* meta.cloud.instance_id -> cloud.instance.id
* meta.cloud.instance_name -> cloud.instance.name
* meta.cloud.machine_type -> cloud.machine.type
* meta.cloud.availability_zone -> cloud.availability_zone
* meta.cloud.project_id -> cloud.project.id
* meta.cloud.region -> cloud.region

Further changes:

* Added alias for old fields
* Update asset generation to include ECS assets in libbeat fields.yml. This is needed for testing as the aliases fields are in there.

### CODE REVIEW

**Comment by webmat** [comment]
Restarted the single Travis test that was failing, and it passed. Likely a transient issue.

Not familiar enough with the provider tests to understand the failure in the Alibaba test:

```
10:44:45 ok  	github.com/elastic/beats/libbeat/outputs/transport/transptest	1.309s	coverage: 47.5% of statements
10:45:06 ok  	github.com/elastic/beats/libbeat/processors/actions	19.241s	coverage: 58.3% of statements
10:45:06 ok  	github.com/elastic/beats/libbeat/processors	19.828s	coverage: 68.3% of statements
10:45:06 command [go test -race -cover -coverprofile /tmp/gotestcover-323955738 github.com/elastic/beats/libbeat/processors/add_cloud_metadata]: exit status 1
10:45:06 --- FAIL: TestRetrieveAlibabaCloudMetadata (19.20s)
10:45:06     <autogenerated>:1: 
10:45:06         	Error Trace:	provider_alibaba_cloud_test.go:85
10:45:06         	Error:      	Not equal: 
10:45:06         	            	expected: common.MapStr{"cloud":common.MapStr{"provider":"ecs", "instance":common.MapStr{"id":"i-wz9g2hqiikg0aliyun2b"}, "region":"cn-shenzhen", "availability_zone":"cn-shenzhen-a"}}
10:45:06         	            	actual  : common.MapStr{}
10:45:06         	            	
10:45:06         	            	Diff:
10:45:06         	            	--- Expected
10:45:06         	            	+++ Actual
10:45:06         	            	@@ -1,2 +1,2 @@
10:45:06         	            	-(common.MapStr) (len=1) {"cloud":{"availability_zone":"cn-shenzhen-a","instance":{"id":"i-wz9g2hqiikg0aliyun2b"},"provider":"ecs","region":"cn-shenzhen"}}
10:45:06         	            	+(common.MapStr) {}
10:45:06         	            	 
10:45:06         	Test:       	TestRetrieveAlibabaCloudMetadata
10:45:06 FAIL
10:45:06 coverage: 85.2% of statements
10:45:06 FAIL	github.com/elastic/beats/libbeat/processors/add_cloud_metadata	19.331s
10:45:10 ok  	github.com/elastic/beats/libbeat/processors/add_host_metadata	1.025s	coverage: 83.1% of statements
```

File: libbeat/processors/add_cloud_metadata/_meta/fields.yml [hunk 1]
```diff
@@ -4,34 +4,32 @@
     Metadata from cloud providers added by the add_cloud_metadata processor.
   fields:
 
-    - name: meta.cloud.provider
```
[the remark this answers is not in the document]
Comment (ruflin) [reply libbeat/processors/add_cloud_metadata/_meta/fields.yml]: fixed

File: dev-tools/ecs-migration.yml [hunk 2]
```diff
@@ -51,6 +51,42 @@
...
+
+- form: meta.cloud.project_id
+  to: cloud.project.id
+  alias: true
+  alias6: true
+
+- form: meta.cloud.region
+  to: cloud.region
+  alias: true
+  alias6: true
```
[the remark this answers is not in the document]
Comment (ruflin) [reply dev-tools/ecs-migration.yml]: Done, not sure how that slipped through above :-(

File: libbeat/processors/add_cloud_metadata/provider_alibaba_cloud_test.go [hunk 3]
```diff
@@ -73,13 +73,13 @@ func TestRetrieveAlibabaCloudMetadata(t *testing.T) {
...
 
 	expected := common.MapStr{
-		"meta": common.MapStr{
-			"cloud": common.MapStr{
-				"provider":          "ecs",
-				"instance_id":       "i-wz9g2hqiikg0aliyun2b",
-				"region":            "cn-shenzhen",
-				"availability_zone": "cn-shenzhen-a",
+		"cloud": common.MapStr{
+			"provider": "ecs",
```
[the remark this answers is not in the document]
Comment (ruflin) [reply libbeat/processors/add_cloud_metadata/provider_alibaba_cloud_test.go]: yep, that one we knew ;-)

**Comment by ruflin** [comment]
@webmat Already tried to reproduce the above failing test locally but couldn't so far. Not sure how it's possible that it fails especially as it's based on local data. Will need to further investigate.

**Comment by webmat** [comment]
About the failing test: is there an actual API call being performed? Or is this based on a static fixture file of the API response?

May just be a network hiccup if there's an API call. I guess we'll see after this test run.

**Pull request merged**